### PR TITLE
add realign and unwarp batch

### PR DESCRIPTION
**Pull request by Remi-Gau**

### CODE REVIEW

**Comment by Remi-Gau** [comment]
Travis is happy and all the "manual tests" pass too so I am merging this (I defo need to automate that... #FamousLastWords)

File: src/spmBatching/setBatchRealign.m [hunk 1]
```diff
@@ -32,31 +46,68 @@
...
+    % -----------------
+    % REALIGN
+    % REALIGN & RESLICE
+    % -----------------
+    %
     % matlabbatch{end}.spm.spatial.realign.estwrite.eoptions.quality = 1;
     % matlabbatch{end}.spm.spatial.realign.estwrite.eoptions.sep = 2;
     % matlabbatch{end}.spm.spatial.realign.estwrite.eoptions.fwhm = 5;
-    % matlabbatch{end}.spm.spatial.realign.estwrite.eoptions.rtm = 1;
+    % matlabbatch{end}.spm.spatial.realign.estwrite.eoptions.rtm = 0;
```
Comment (CerenB) [inline src/spmBatching/setBatchRealign.m:80]: @Remi-Gau uhmm I do not know how you use this spm_defaults + this section. Is it for visualisation of what is set in spm_defaults or is it the opposite? Also, I'm a bit lost (not remembering what I was doing previously), what was in the previous form of cpp-pipeline? This was set to 1 and now you are changing it to 0? And how is it in spm_defaults?

File: src/spmBatching/setBatchRealign.m [hunk 2]
```diff
@@ -32,31 +46,68 @@
...
+    %   matlabbatch{1}.spm.spatial.realignunwarp.uweoptions.sot = [];
+    %   matlabbatch{1}.spm.spatial.realignunwarp.uweoptions.uwfwhm = 4;
+    %   matlabbatch{1}.spm.spatial.realignunwarp.uweoptions.rem = 1;
+    %   matlabbatch{1}.spm.spatial.realignunwarp.uweoptions.noi = 5;
+    %   matlabbatch{1}.spm.spatial.realignunwarp.uweoptions.expround = 'Average';
+    %
+    %   matlabbatch{1}.spm.spatial.realignunwarp.uwroptions.rinterp = 3;
+    %   matlabbatch{1}.spm.spatial.realignunwarp.uwroptions.wrap = [0 0 0];
+    %   matlabbatch{1}.spm.spatial.realignunwarp.uwroptions.mask = 1;
+
```
Comment (CerenB) [inline src/spmBatching/setBatchRealign.m:112]: realign & unwarp is.... now set for native space?

**Review by CerenB — COMMENTED** [review]
sorry to be late with this review. This is also not code review, I simply want to ask is it now by default pipeline realign + unwrap instead of realign + reslice?

**Comment by Remi-Gau** [comment]
> sorry to be late with this review. This is also not code review, I simply want to ask is it now by default pipeline realign + unwrap instead of realign + reslice?

Nope: the default pipeline will stick to a simple "realign". For now.

I am ok to make the realign and unwarp the default, but I think this should be discussed in a separate issue and see how other fMRI-ers in the lab feel about that.

File: src/spmBatching/setBatchRealign.m (hunk 1 above)
Comment (Remi-Gau) [reply src/spmBatching/setBatchRealign.m:80]: Yes this is only for "users" to see what the other defaults of the pipelines are.

Currently, the defaults of the pipeline are set in [src/spmDefautls](https://github.com/cpp-lln-lab/CPP_BIDS_SPM_pipeline/tree/dev/src/spmDefaults).

Any users can "override" those by having their own "spm_defaults" in the matlab path.
I actually need to :
- document that better
- make sure that the order in which those `spm_defaults` appear in path is the correct one (to make sure that the user's default takes precedence over those of the pipeline.) 

When in doubt about what SPM uses as defaults, you can type `edit spm_defaults`. :wink: The file also comes with a "how to use this" section. :-) 

So for realign the `realign to mean` (rtm) is set to `true` but it is false `for realign and unwarp`.

I am just setting this to `false` for all the cases of realingment we use.

Does that make sense?

File: src/spmBatching/setBatchRealign.m (hunk 2 above)
Comment (Remi-Gau) [reply src/spmBatching/setBatchRealign.m:112]: I am not sure what you mean. :thinking:

Comment (CerenB) [reply src/spmBatching/setBatchRealign.m:112]: my bad.. my mind went to something else!

**Pull request merged**